sw/device/lib/base/mmio.h

```diff
@@ -37,9 +48,13 @@ inline reg32_t reg32_from_addr(uintptr_t address) {
  * @param offset the offset to read at, in bytes.
  * @return the read word.
  */
+#ifndef MOCK_MMIO
```

Comment:
Maybe have a single ifdef for both functions, something like this:
```
#ifndef MOCK_MMIO
/**
 * description
 */
inline uint32_t reg32_read(reg32_t base, ptrdiff_t offset) {
  return ((volatile uint32_t *)base.inner_ptr)[offset / sizeof(uint32_t)];

/**
 * description
 */
inline void reg32_write(reg32_t base, ptrdiff_t offset, uint32_t value) {
  ((volatile uint32_t *)base.inner_ptr)[offset / sizeof(uint32_t)] = value;
}
#else // MOCK_MMIO
uint32_t reg32_read(reg32_t base, ptrdiff_t offset);
void reg32_write(reg32_t base, ptrdiff_t offset, uint32_t value);
#endif // MOCK_MMIO
```
What do you think? To me it seems a little cleaner.

Reply:
Sure, we can do that.